### PR TITLE
Migrate some end-to-end test to use test builders 🎧

### PR DESCRIPTION
**Pull request by vdemeester**
This convert most of the `test/*_test.go` to using builders.

Marking this as WIP as I'm pretty sure I did some typos :upside_down_face: 

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

### CODE REVIEW

**Comment by vdemeester** [comment]
oh yay it's green :green_heart: Keeping the `wip` as I need to update some tests :angel:

**Comment by vdemeester** [comment]
Rebased :angel:

File: test/kaniko_task_test.go [hunk 1]
```diff
@@ -97,86 +87,38 @@ func createSecret(c *knativetest.KubeClient, namespace string) (bool, error) {
...
-				VolumeSource: corev1.VolumeSource{
-					Secret: &corev1.SecretVolumeSource{
-						SecretName: "kaniko-secret",
-					},
-				},
+		stepOps = append(stepOps,
+			tb.VolumeMount(corev1.VolumeMount{Name: "kaniko-secret", MountPath: "/secrets"}),
+			tb.EnvVar("GOOGLE_APPLICATION_CREDENTIALS", "/secrets/config.json"),
+		)
+		taskSpecOps = append(taskSpecOps, tb.TaskVolume("kanikou-secret", tb.VolumeSource(corev1.VolumeSource{
```
Comment (adrcunha) [inline test/kaniko_task_test.go]: Shouldn't it be `kaniko-secret`?

Comment (vdemeester) [reply test/kaniko_task_test.go]: @adrcunha indeed :joy_cat:

File: test/cluster_resource_test.go [hunk 2]
```diff
@@ -70,34 +71,14 @@ func TestClusterResource(t *testing.T) {
...
-			}},
-		},
-	}
+	return tb.PipelineResource(name, namespace, tb.PipelineResourceSpec(
+		v1alpha1.PipelineResourceTypeCluster,
+		tb.PipelineResourceSpecParam("Url", "https://1.1.1.1"),
+		tb.PipelineResourceSpecParam("username", "test-user"),
+		tb.PipelineResourceSpecParam("password", "test-password"),
+		tb.PipelineResourceSpecSecretParam("cadata", sname, "cadatakey"),
+		tb.PipelineResourceSpecSecretParam("token", sname, "tokenkey"),
```
Comment (bobcatfish) [inline test/cluster_resource_test.go]: niiiiiiice

**Comment by bobcatfish** [comment]
hm looks like a bunch of weird failures :thinking: let me know if you think these are bogus @vdemeester 

/test pull-knative-build-pipeline-integration-tests

**Comment by vdemeester** [comment]
> hm looks like a bunch of weird failures thinking let me know if you think these are bogus @vdemeester
> 
> /test pull-knative-build-pipeline-integration-tests

@bobcatfish I am guessing I messed up when rebasing… let's see :upside_down_face:

**Comment by vdemeester** [comment]
Interesting the failure seems to be cause by the following

```
I0108 11:02:42.110]   taskRuns:
I0108 11:02:42.110]     helloworld-pipelinerun1-create-file-kritis:
I0108 11:02:42.110]       conditions:
I0108 11:02:42.111]       - lastTransitionTime: "2019-01-08T10:53:10Z"
I0108 11:02:42.111]         message: 'pod status "PodScheduled":"False"; message: "0/3 nodes are available:
I0108 11:02:42.111]           3 node(s) had no available volume zone."'
I0108 11:02:42.111]         reason: Pending
I0108 11:02:42.111]         status: Unknown
I0108 11:02:42.111]         type: Succeeded
I0108 11:02:42.112]       podName: helloworld-pipelinerun1-create-file-kritis-pod-dce3ed
I0108 11:02:42.112]       startTime: "2019-01-08T10:52:43Z"
```

**Comment by vdemeester** [comment]
Rebase :angel:

**Pull request merged**